flume-ng-sinks/flume-hdfs-sink/src/test/java/org/apache/flume/sink/hdfs/TestBucketWriter.java

```diff
@@ -374,6 +376,45 @@ public void testInUseSuffix() throws IOException, InterruptedException {
   }
 
   @Test
+  public void testTranslatedInUseSuffix() throws IOException, InterruptedException {
+    final int ROLL_INTERVAL = 1000; // seconds. Make sure it doesn't change in course of test
+    final String SUFFIX = "JAI_MATA_DI_%Y%m%d";
+    
+    // Need to override system time use for test so we know what to expect
+    final long testTime = System.currentTimeMillis();
+
+    Clock testClock = new Clock() {
+      public long currentTimeMillis() {
+        return testTime;
+      }
+    };
+    Clock prevClock = BucketPath.getClock();
+    
+    BucketPath.setClock(testClock);
+    
+    MockHDFSWriter hdfsWriter = new MockHDFSWriter();
+    
+
+    Event e = EventBuilder.withBody("foo", Charsets.UTF_8);
+    e.getHeaders().put("timestamp", String.valueOf(testTime));
+
+    String translatedSuffixString = BucketPath.escapeString(SUFFIX, e.getHeaders());
+    
+    BucketWriter bucketWriter = new BucketWriter(
+        ROLL_INTERVAL, 0, 0, 0, ctx, "/tmp", "file", "", translatedSuffixString, null, null,
```

Comment:
No, BucketWriter class doesnt internally do translation of suffix it gets the translated suffix and append it. see below code from BucketWriter at line number 240. Also I followed the method use for the other two (realPath, realName) to be consistent. 

        if (fileSuffix != null && fileSuffix.length() > 0) {
          fullFileName += fileSuffix;
        } else if (codeC != null) {
          fullFileName += codeC.getDefaultExtension();
        }